airbyte-cdk/python/airbyte_cdk/sources/embedded/tools.py

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Callable, Dict, Iterable, Optional
+
+from airbyte_cdk.models import AirbyteStream
+
+
+def get_first(iterable: Iterable[Any], predicate: Callable[[Any], bool] = lambda m: True) -> Optional[Any]:
+    return next(filter(predicate, iterable), None)
+
+
+def get_defined_id(stream: AirbyteStream, data: Dict[str, Any]) -> Optional[str]:
+    import dpath
```

Comment:
why is dpath imported from the method instead of at the top of the file?

Reply:
Fixed